docs/reference/elasticsearch/mapping-reference/semantic-text-reference.md

```diff
@@ -157,6 +157,88 @@ You can [pre-chunk content](./semantic-text-ingestions.md#pre-chunking) by provi
 Refer to the [{{infer-cap}} API documentation](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-inference-put#operation-inference-put-body-application-json-chunking_settings) for values for `chunking_settings` and to [Configuring chunking](docs-content://explore-analyze/elastic-inference/inference-api.md#infer-chunking-config) to learn about different chunking strategies.
 
 
+## Automatic pre-filtering for dense vector embeddings [auto-prefiltering]
+```{applies_to}
+stack: ga 9.3
+serverless: ga
+```
+
+Querying `semantic_text` fields that have dense vector embeddings automatically applies
+filters found in the Query DSL tree or [ES|QL](/reference/query-languages/esql.md) query
+as pre-filters in order to ensure the requested number of results is returned.
+
+In the following example we are using a `match` query against a `semantic_text` field
+in order to find the 10 most relevant documents that match the query "quick drying t-shirts".
+However, we are only interested in green t-shirts, so we add a filter query to exclude other colors.
+
+In Query DSL we would express this query as follows:
+
+
+```console
+POST my-index/_search
+{
+  "size" : 10,
+  "query" : {
+    "bool" : {
+      "must" : {
+        "match": {
+          "dense_semantic_text_field": {
+            "query": "quick drying t-shirts"
+          }
+        }
+      },
+      "filter" : {
+        "term" : {
+          "color": {
+            "value": "green"
+          }
+        }
+      }
+    }
+  }
+}
+```
+% TEST[skip:Requires {{infer}} endpoint]
+
+
+The `term` query will be applied as a pre-filter, meaning that when the *knn* search executes on
```

Comment:
I'm making the assumption that people reading this understand that a `match` query against a dense `semantic_text` field is doing `knn` search under the hood. I wonder if we need to add more explanation here or not.

Reply:
I think so. Maybe we don't need to get technical - we can just say that in case it's needed, it will be applied as a pre-filter so we keep the expected number of results back.